src/Helper/HeaderHelper.php

```diff
@@ -17,16 +17,38 @@ public static function getValueAndParameters(string $headerValue): array
         if ($headerValue === '') {
             return [];
         }
-        $parts = preg_split('/\s*;\s*/', $headerValue, -1, PREG_SPLIT_NO_EMPTY);
-        $output = [array_shift($parts)];
-        foreach ($parts as $part) {
-            [$key, $headerValue] = explode('=', $part, 2);
-            if (preg_match('/^"(?<value>.*)"$/', $headerValue, $matches) === 1) {
-                $headerValue = $matches['value'];
-            }
-            $output[$key] = $headerValue;
+        $parts = preg_split('/\s*;\s*/', $headerValue, 2, PREG_SPLIT_NO_EMPTY);
+        $output = [$parts[0]];
+        if (count($parts) === 1) {
+            return $output;
         }
-        return $output;
+        return array_merge($output, self::getParameters($parts[1]));
+    }
+
+    /**
+     * Explode header value to parameters (eg. q=2;version=6)
+     *
+     * @link https://tools.ietf.org/html/rfc7230#section-3.2.6
+     */
+    public static function getParameters(string $headerValue): array
+    {
+        $headerValue = trim($headerValue);
+        if ($headerValue === '') {
+            return [];
+        }
+        $output = new \stdClass();
```

Comment:
I think it should be simply ```$output = [];```

Reply:
https://github.com/yiisoft/yii-web/commit/8881b0421485031976912f5c4cc8321b8dde0468